sql/planbuilder/from.go

```diff
@@ -664,49 +661,104 @@ func (b *Builder) buildResolvedTable(inScope *scope, db, schema, name string, as
 		if !ok {
 			b.handleErr(sql.ErrDatabaseNotFound.New(schema))
 		}
-	}
 
-	if view := b.resolveView(name, database, asOfLit); view != nil {
-		outScope.node = view
-		tabId := outScope.addTable(strings.ToLower(view.Schema()[0].Name))
-		var cols sql.ColSet
-		for _, c := range view.Schema() {
-			id := outScope.newColumn(scopeColumn{
-				db:          db,
-				table:       name,
-				col:         strings.ToLower(c.Name),
-				originalCol: c.Name,
-				typ:         c.Type,
-				nullable:    c.Nullable,
-			})
-			cols.Add(sql.ColumnId(id))
-		}
-		if tin, ok := view.(plan.TableIdNode); ok {
-			// TODO should *sql.View implement TableIdNode?
-			outScope.node = tin.WithId(tabId).WithColumns(cols)
+		if view := b.resolveView(name, database, asOfLit); view != nil {
+			return resolvedViewScope(outScope, view, db, name)
 		}
 
-		return outScope, true
-	}
+		if asOfLit != nil {
+			tab, database, tableResolveErr = b.cat.DatabaseTableAsOf(b.ctx, database, name, asOfLit)
+		} else {
+			tab, _, tableResolveErr = b.cat.DatabaseTable(b.ctx, database, name)
+		}
+	} else if db != "" {
+		if asOfLit != nil {
+			tab, database, tableResolveErr = b.cat.TableAsOf(b.ctx, db, name, asof)
+		} else {
+			tab, database, tableResolveErr = b.cat.Table(b.ctx, db, name)
+		}
+	} else if schema != "" {
+		// TODO: revise this
```

Comment:
this is getting bad enough that it might make sense to have a doc for all the table name binding edge cases

Reply:
I reworked this significantly, will do another pass on this as I figure out the right package separation.